application/app/src/main/java/com/example/buber/Services/ApplicationService.java

```diff
@@ -21,4 +26,56 @@ public static User createNewUser(
             String phoneNumber
     ) { return null; }
 
+
+//    --------------------MADEEHA---------------------------------------
+    /*TODO:
+       //get username and password from UI when user creates an account
+       //append @Uber.com to the username
+       //call my create account function in LoginAuth:              LoginAuth.createAccount(username@uber.com, password, (method called on sucess))
+                    To finish creating the account we need from UI:
+                    - is it a Rider.class or driver.class? or both?
+                       Get the document ID:  LoginAuth.getcurrentUserDocID()
+                      and Create : DatabaseManager.createRider( document ID) or   DatabaseManager.createDriver( document ID)                //here we created an empty document in the rider or driver collection
+                   Next populate/Create a rider or Driver class object (extend user):
+                    User class has:
+                    -Contact                                           //I commented out cause im not sure about it
+                    -Location.class of the rider or driver:             Location(private float latitude, private float longitude);
+                    -Account.class                                      Account (firstName, lastName, email(getcurrentUser().getEmail());
+                    -String username;                                   String getcurrentUser().getEmail()- minus @uber.com(Email is the username with @uber.com appended)
+                  Rider has no other attributes
+                  Driver has ratting?
+            Once the rider or driver class has been created update the driver or rider with data                                                  //here we populate the empty document in the rider or driver collection
+                updateDriver(Document ID)
+                ***cast the Driver as a user object;
+                now we have a current user
+            }
+
+     */
+    //--------------------------------------------Testing------------------------------------------
+    private AuthDBManager login = new AuthDBManager();
+    private DBManager DB = new DBManager();
+
+    public void createAccountService(){
+        /*TODO: TODO: Get username and password from UI when user creates an account*/
+        login.createAccount("madeeha@BUber.com", "123456" , this);    // If account is created aftersuccessfulCreataAccount() is called
+    }
+
+
+    public User aftersuccessfulCreataAccount(String DocID){
+        //- is it a Rider.class or driver.class? or both?
+        Rider riderObj = new Rider();
+        Driver driverObj = new Driver();
+
+        DB.createDriver(DocID, driverObj); //here we created an empty document in the rider or driver collection
```

Comment:
Just delete the two method calls to create the rider/driver for now, lets keep it as simple as possible and grow the functionality over time.

Reply:
Return a new User object with the attributes that are passed down from the controller. See the controller code.
![image](https://user-images.githubusercontent.com/34895471/75099287-251aae80-557d-11ea-9ff6-9ee66e398ee5.png)

Reply:
So yeah this makes sense but since Driver and Rider extend user. I might be wrong but in fb we need to create a rider or driver with the same DocID as fbAuth user so that when it returns we can upcast rider or driver to a user